client/blocks/app-banner/index.jsx

```diff
@@ -131,9 +132,21 @@ class AppBanner extends Component {
 			}
 		}
 
-		//TODO: update when deferred deep links are available
 		if ( this.isiOS() ) {
-			return 'itms://itunes.apple.com/us/app/wordpress/id335703880?mt=8';
+			switch ( currentSection ) {
+				case EDITOR:
+					return 'https://apps.wordpress.com/get#post';
+				case NOTES:
+					return 'https://apps.wordpress.com/get#notifications';
+				case READER:
+					if ( currentPathFragment.length === 0 ) {
```

Comment:
why are we special-casing the Reader with the fragment length check?

Reply:
The Reader doesn't have its own path when you view it on the web – so the root of the Reader just sits at `WordPress.com`. Each of the other routes have their own distinct path – `/stats`, `/notifications`, `/post`. The Reader can also be accessed at `/read`, but this handles the case where you're just at the root and there's no path, and ensures the app knows which section to take the user to.

Reply:
hm. it's still a bit unclear how this is supposed to function. if Reader is a special exception then I think it begs for a strong explanatory comment otherwise someone will come here and trash the code at some point in the future unless it's obvious why it's there :wink:

Reply:
Definitely a comment makes sense. I'll try and explain here a bit more clearly too with a couple of examples:

* If you visit the editor in a mobile web browser, the URL is `wordpress.com/post`, and you'll see the editor app banner. If the app is launched from here, we handle the path `/post` and display the editor.
* If you visit Discover in the Reader in a mobile web browser, the URL is `wordpress.com/discover` and you'll see the Reader app banner. If the app is launched from here, we handle the path `/discover`, and display the Discover feed in the app's Reader.
* If you visit the Reader (tap the Reader icon in the top bar) in a mobile web browser, the URL is just `wordpress.com`, and you'll see the Reader app banner. If the app is launched from here, without a special case, we'd just get the path `/`, and have no way to differentiate it from the logged out WordPress.com URL (which we don't want to deep link into the app). Hence, a special case to interpret this as `/read` in the app.

Reply:
A one line comment should be sufficient. We basically want WordPress.com root or `/read` in this case?

Reply:
Essentially – I've added an explanatory comment to the code.